commands/report_generate.js

```diff
@@ -0,0 +1,83 @@
+const request = require("request")
+const constants = require("./utils/constants.js")
+const process = require("process")
+
+
+function generate_report(args) {
+
+    return new Promise(function (resolve, reject) {
+        var username = ""
+        var access_key = ""
+        //Check for username
+        if ("user" in args) {
+            username = args.user
+        } else if (process.env.LT_USERNAME) {
+            console.log("Setting Username from ENV", process.env.LT_USERNAME)
+            username = process.env.LT_USERNAME
+        } else {
+            reject("Username not provided")
+        }
+
+        //Check for access key
+        if ("access_key" in args) {
+            access_key = args.access_key
+        } else if (process.env.LT_ACCESS_KEY) {
+            console.log("Setting Access Key from ENV", process.env.LT_ACCESS_KEY)
+            access_key = process.env.LT_ACCESS_KEY
+        } else {
+            reject("Access Key not provided")
+        }
+
+        if (!("buildId" in args)) {
+            reject("Please provide a build ID")
+        }
+        var env = "prod"
+        if ("env" in args) {
+
+            if (["stage", "prod", "beta"].includes(args["env"])) {
+                env = args["env"]
+
+            } else {
+                console.log("Environment can be stage, beta or prod, setting Env to prod")
+
+            }
+
+        }
+      
+        const authorizationHeaderValue = Buffer.from(username + ':'+ access_key).toString('base64');
+        let options = {
+            url: constants[env].REPORT_GENERATE_URL,
+            headers: {
+                'Authorization': 'Basic ' + authorizationHeaderValue
+            },
+            body: JSON.stringify({
+                "build_id": Number(args.buildId)
+            }),
+        }
+        request.post(options , function(err, res, body) {
+            if (err) {
+                reject(err)
+            }
+            if (res.statusCode == "401") {
```

Comment:
Either all `res.statusCode`  condition should be check in string format or in number.
Mixed used of string and number conditional checking can cause unpredictable bug or issue.

```js 
if (res.statusCode === 401) 
```